docs/learn/smart-contract-developers/create-lsp7-token.md

```diff
@@ -62,217 +79,346 @@ contract CustomToken is LSP7Mintable, LSP7Burnable {
+:::tip
 
-#### Deploy using a Universal Profile (Recommended)
+Add the `--verbose` and `--show-stack-traces` flags to get further debugging information and gas data.
 
-Deploy your contract using your Universal Profile. Make sure to have the [Universal Profile Browser Extension](/install-up-browser-extension) installed.
+This is especially important if a lot of functionality is inherited, as the bytecode might succeed the EVM limit of 24k bytes. You can find more information about optimization settings within the [Getting Started](./getting-started.md) section.
 
-:::note
+:::
 
-The `privateKey` coming from your UP extension is the private key of the EOA that controls your UP (more information about controllers can be found in the [Key Manager](../../standards/universal-profile/lsp6-key-manager.md) page). This address will need to be funded using the [Testnet Faucet](https://faucet.testnet.lukso.network/).
+## Deploy the Token
 
-:::
-Create the script that will deploy the contract as your Universal Profile.
+After the contract file has been successfully compiled, you are ready to create a script to deploy it's token on the [LUKSO Testnet network](../../networks/testnet/parameters). To deploy your token on chain we **recommend using a controller and address of a Universal Profile**, so your asset will be connected and fetchable from your on-chain persona. Optionally, you can also use a regular Externally Owned Account (EOA).
 
-1. Add your Universal Profile address to your `.env` file in `UP_ADDR`
-2. Load the associated UP
-3. Get the bytecode of your contract
-4. Use `staticCall` method to get the address of the contract
-5. Deploy the contract
+<Tabs groupId="deployment">
+  <TabItem value="up" label="Deploy with Universal Profile">
 
-```ts title="scripts/deployUP.ts"
+```ts title="scripts/deployMyCustomToken.ts"
+import hre from 'hardhat';
 import { ethers } from 'hardhat';
 import * as dotenv from 'dotenv';
-import UniversalProfileArtifact from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
-import { CustomToken__factory } from '../typechain-types';
+import LSP0Artifact from '@lukso/lsp-smart-contracts/artifacts/LSP0ERC725Account.json';
 
-// load env vars
+// Load the environment variables
 dotenv.config();
-const { UP_ADDR } = process.env;
 
-async function main() {
-  // load the associated UP
-  const universalProfile = await ethers.getContractAtFromArtifact(
-    UniversalProfileArtifact,
-    UP_ADDR as string,
+async function deployToken() {
+  // Setup the provider
+  const provider = new ethers.JsonRpcProvider(
+    'https://rpc.testnet.lukso.gateway.fm',
+  );
+
+  // Setup the controller used to sign the deployment
+  const signer = new ethers.Wallet(process.env.PRIVATE_KEY as string, provider);
+  console.log(
+    'Deploying contracts with Universal Profile Controller: ',
+    signer.address,
```

Comment:
we can do it this way but this we are in a hardhat environment 
may be we could use the ` const [deployer] = await ethers.getSigners()` as long as we have updated our `hardhat.config.ts` file

Reply:
Thanks for this! Indeed we should use `ethers.getSigners()`- we already use it for EOA scripts, but just not for UPs... i simply copied it over from the `lukso-playground` - will change on both ends 👍🏻